src/rdb.c

```diff
@@ -1943,8 +1944,14 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key) {
                 }
                 o->type = OBJ_HASH;
                 o->encoding = OBJ_ENCODING_ZIPLIST;
-                if (hashTypeLength(o) > server.hash_max_ziplist_entries)
+                if (hashTypeLength(o) > server.hash_max_ziplist_entries) {
                     hashTypeConvert(o, OBJ_ENCODING_HT);
+                } else if (server.rdb_convert_ziplist || deep_integrity_validation) {
```

Comment:
i think that when i wrote that comment i wanted to let the fuzzer (and existing consistent tests in corrupt-dump.tcl) to be able to run commands on a database full with ziplists. so that if we have some issue in for instance HRANDMEMBER that works on a corrupt ziplist, it'll still be able to find it.

but now that i think of it, these issues can still be uncovered when the deep mode is disabled, in which case we won't run the listpack conversion, so the tests can still get to test ziplists.